### PR TITLE
feat(reward): read a frozen VLM's token log-probability as a world-model reward

### PR DESCRIPTION
**Pull request by ZJLi2013**
### Description

Add `TOPRewardModel`, which scores generated frames with a frozen VLM by reading `log P(" True")` after a prompt that shows the frames and then *asserts* the task was completed, following TOPReward. It returns a 0/1 label rather than the probability, so the label means the same thing as the ResNet reward model's `round()` and drives `terminations` and the loss mask unchanged -- no new aggregation code downstream.

Wired into the Wan world-model env through the existing `cfg.reward_model.type` dispatch, plus env presets and recipes for LIBERO-spatial and LIBERO-object.

### Motivation and Context

#1049 added the VLM reward path, but it scores by `model.generate` plus parsing the decoded text. 

ResnetRewardModel is trained on the simulator's privileged state, so a world model with no simulator behind it cannot obtain one, and every task suite needs its own copy. A frozen VLM needs neither: switching suites only changes the instruction text.

### How has this been tested?

`Wan + OpenVLA-OFT + GRPO` on two LIBERO suites, 8 x MI300X, 20 steps each, every arm paired with a ResNet arm differing only in the reward model. Scored on the real LIBERO simulator with all 500 episodes enumerated: one reading has a standard error of 2.2 points, a difference of two readings 3.1 points.

| suite | base | ResNet classifier | frozen Qwen3-VL |
|---|---|---|---|
| spatial | `0.448` | `0.574` | `0.560` |
| object | `0.342` | `0.368` | `0.348` |

Best point of each arm; spatial peaks at step 10 for the VLM and step 20 for the classifier, object at step 10 and step 20 respectively.

**The frozen VLM matches a classifier trained on the exact domain, on both suites.** 
**The frozen scorer's in-world-model reading tracks the real simulator; the classifier's does not.** On object at step 20:

| scorer | in-world-model | real simulator | 
|---|---|---|
| ResNet classifier | 0.55 | 0.348 |
| frozen Qwen3-VL | 0.35 | 0.348 |

Five unit tests with the VLM forward stubbed out, no GPU needed: 

```
tests/unit_tests/test_topreward_model.py .....                       [100%]
```

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: rlinf/models/embodiment/reward/topreward_model.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            .numpy()
+            .reshape(num_envs, self.chunk, *images.shape[1:])
+        )
+
+        labels = torch.zeros(num_envs, dtype=images.dtype, device=images.device)
+        for env in range(num_envs):
+            chunk = frames[env]
+            # A chunk scored on its own loses the separation between succeeding and
+            # failing episodes, so the call also sees how the episode got here.
+            past = self._history.get(env)
```
[the remark this answers is not in the document]
Comment (ZJLi2013) [reply rlinf/models/embodiment/reward/topreward_model.py:153]: fixed and pushed

File: docs/source-zh/rst_source/examples/embodied/wan.rst [hunk 2]
```diff
@@ -263,6 +263,75 @@ OpenVLA-OFT + GRPO 使用 ``examples/embodiment/config/wan_libero_spatial_grpo_o
...
+
+每个 Wan checkpoint 自带的 ``ResnetRewModel`` 是在模拟器的特权状态上训练的，因此每个套件都需要
+一份自己的 ``resnet_rm.pth``。``TOPRewardModel`` 用一个全程不训练的 VLM 取代它：把生成的画面连同
+一句\ **断言任务已完成**\ 的话一起喂进去，再读这句话的 ``log P(" True")``。换套件只需换指令文本。
+
+判分器需要 ``transformers >= 4.57`` 才认得 Qwen3-VL，比 OpenVLA-OFT 钉的版本新，因此装到独立环境：
+
+.. code:: bash
+
+   bash requirements/install.sh embodied --model qwen3_vl --env wan
```
[the remark this answers is not in the document]
Comment (ZJLi2013) [reply docs/source-zh/rst_source/examples/embodied/wan.rst]: fixed and pushed

File: docs/source-en/rst_source/examples/embodied/wan.rst [hunk 3]
```diff
@@ -266,6 +266,84 @@ OpenVLA-OFT + GRPO uses ``examples/embodiment/config/wan_libero_spatial_grpo_ope
...
+it with a VLM that is never trained: it shows the generated frames together with a sentence
+*asserting* the task was completed, then reads ``log P(" True")`` for that sentence. Switching
+suites changes only the instruction text.
+
+The scorer needs ``transformers >= 4.57`` for Qwen3-VL, which is newer than the version
+OpenVLA-OFT pins, so install it into its own environment:
+
+.. code:: bash
+
+   bash requirements/install.sh embodied --model qwen3_vl --env wan
```
[the remark this answers is not in the document]
Comment (ZJLi2013) [reply docs/source-en/rst_source/examples/embodied/wan.rst]: fixed and pushed
